### PR TITLE
Fix/wiki 1270

### PR DESCRIPTION
**Pull request by aboughzela**

### CODE REVIEW

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java [hunk 1]
```diff
@@ -50,10 +52,27 @@
...
+  private String  nodeName = null;
+
   private final static String INNER_OBJECT_SPLIT = ":";
   private final static String OUTER_OBJECT_SPLIT = ";";
 
-  public WikiMigrationSettingService(SettingService settingService) {
+  public WikiMigrationSettingService(SettingService  settingService, InitParams initParams) {
+    //Init cluster parameters
+    if (initParams != null) {
+      ValueParam param = initParams.getValueParam(CLUSTER_MODE_PARAM);
```
Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: Do we really need this param ? We could set the setting in any case (so even in non clustered mode).

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java [hunk 2]
```diff
@@ -50,10 +52,27 @@
...
+      if (param != null) {
+        clusterMode = "true".equalsIgnoreCase(param.getValue());
+      }
+      param = initParams.getValueParam(NODE_MODE_PARAM);
```
Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: Is it useful to expose this as a param of the service ? We could directly get the exo.cluster.node.name property.

Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: Does exo.cluster.node.name have a default default when we are not in cluster ?

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/MigrationService.java [hunk 3]
```diff
@@ -139,6 +145,16 @@ public void start() {
         //Let's start the migration
         LOG.info("=== Start Wiki data migration from JCR to RDBMS");
 
+        //Unregister node name , If  System.exit() is called before Thread migration is done.
+        if(settingService.isClusterMode()) {
+          Runtime.getRuntime().addShutdownHook(new Thread("WIKI-MIGRATION-HOOK") {
```
Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/MigrationService.java]: This has not been added for social migration, it should IMO.

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java (hunk 1 above)
Comment (aboughzela) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: Do you mean , remove this param and use directly exo.profiles sytem property ?
I have used the same logic as social migration (homogeneity of code)

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java (hunk 2 above)
Comment (aboughzela) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: this param exist only on cluster mode

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java (hunk 1 above)
Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/WikiMigrationSettingService.java]: No I mean I think we don't need this param at all, we can set the setting in any case, no matter if we are in cluster or not (in non cluster mode this setting will simply be not used).

File: wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/MigrationService.java (hunk 3 above)
Comment (thomasdelhomenie) [inline wiki-jpa-migration/src/main/java/org/exoplatform/wiki/jpa/migration/MigrationService.java]: Yes. I think we will wait for the end of this fix, then we will open an issue to improve the process in social migration.

**Comment by thomasdelhomenie** [comment]
Merged

**Pull request closed without merging**